File: Medium/2406. Divide Intervals Into Minimum Number of Groups.java
import java.util.Arrays;
import java.util.PriorityQueue;

class Solution {
    public int minGroups(int[][] intervals) {
        Arrays.sort(intervals, (x, y) -> Integer.compare(x[0], y[0]));
        final var minHeap = new PriorityQueue<Integer>();
        for (var interval : intervals) {
            if (!minHeap.isEmpty() && minHeap.peek() < interval[0]) {
                minHeap.poll();
            }
            minHeap.offer(interval[1]);
        }
        return minHeap.size();
    }
}

File: Medium/2406. Divide Intervals Into Minimum Number of Groups.py
from ast import List
from heapq import heappop, heappush

class Solution:
    def minGroups(self, intervals: List[List[int]]) -> int:
        min_heap = []
        for start, end in sorted(intervals, key = lambda interval: interval[0]):
            if min_heap and min_heap[0] < start:
                heappop(min_heap)
            heappush(min_heap, end)
        return len(min_heap)
        